src/ansys/hps/client/client.py

```diff
@@ -190,6 +191,13 @@ def __init__(
                 verify=self.verify,
             )
             self.access_token = tokens["access_token"]
+
+            try:
+                parsed_jwt = jwt.decode(self.access_token, options={"verify_signature": False})
+                self.username = parsed_jwt.get("preferred_username", self.username)
```

Comment:
@wehrler If the name can be extracted from the token should we rather compare it to the username set on the client object and throw if there's a mismatch?

Reply:
I agree. If the client had the username intentionally set then it would not be ideal to silently change it.

*See latest changes.